Bump Elixir/Erlang and fix warnings

sviik opened the pull request.

indrekj commented on CHANGELOG.md:
```diff
@@ -1,5 +1,9 @@
 # Changelog
 
+## 2.0.0
```
are there breaking changes for the major bump?

sviik replied:
Actually no. I figured Elixir/OTP update would be worthy of a major bump but now when I think of it, the changes don't break anything for earlier versions. Will change the version number.

sviik commented:
@indrekj re-requesting review:
1. Changed version number https://github.com/salemove/logstash_logger_formatter/pull/17/commits/9e7695eeac2afe0286a611de47f2410e3d7da396
2. Added a fix for file metadata value f1e1fe395e24aea9d5d4235a740afce87ed8f320

Also, GH is waiting for a status from `OTP 23.2 / Elixir 1.11` but it will never arrive as these versions are not in the test matrix anymore. Do you know if I should re-create a PR or is there something else that needs to be done?

indrekj commented:
> I should re-create a PR or is there something else that needs to be done

Somebody with admin permissions could remove that requirement, I myself don't have permissions for this repo though

bautrey37 commented:
Thanks for fixing! I was about to look into the warnings from this library.
